src/main/java/core/basesyntax/figures/Square.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.figures;
+
+import java.util.Locale;
+
+public class Square extends Figure {
+    private double side;
+    private double area;
+
+    public Square(String color, double side) {
+        super(color);
+        this.side = side;
+        this.area = calculateArea();
+    }
+
+    public double getSide() {
+        return side;
+    }
+
+    public void setSide(double side) {
+        this.side = side;
+        this.area = calculateArea();
```

Comment:
Similar to the constructor, avoid recalculating and storing the area in the setter. The 'getArea()' method should handle the calculation.